In [1]:
import numpy as np 
import pandas as pd 

In [2]:
df = pd.read_csv(r"C:\TECH FEST\archive sensor data\agriculture_dataset.csv")

In [3]:
print(df.head())
print(df.shape)
print(df.columns)

   High_Resolution_RGB  Multispectral_Images  Thermal_Images  Temporal_Images  \
0                    0                     0               0                0   
1                    1                     1               0                0   
2                    0                     0               0                0   
3                    0                     0               0                0   
4                    0                     1               1                1   

   Spatial_Resolution  GPS_Coordinates  Field_Boundaries  Elevation_Data  \
0            0.667324           201538                 3       28.207634   
1            1.459000           215854                 3       82.335147   
2            0.500442           890802                 3       83.865629   
3            1.865161           605584                 3       20.747905   
4            1.392331           871732                 3       22.588815   

   Canopy_Coverage      NDVI  ...  Weed_Coverage  Pest_D

In [4]:
# Original DataFrame
print(df.columns)
# 32 columns

# Keep only 4 columns
df = df[['Temperature', 'Humidity', 'Soil_Moisture', 'Crop_Health_Label']]

print(df.columns)

Index(['High_Resolution_RGB', 'Multispectral_Images', 'Thermal_Images',
       'Temporal_Images', 'Spatial_Resolution', 'GPS_Coordinates',
       'Field_Boundaries', 'Elevation_Data', 'Canopy_Coverage', 'NDVI', 'SAVI',
       'Chlorophyll_Content', 'Leaf_Area_Index', 'Crop_Stress_Indicator',
       'Temperature', 'Humidity', 'Rainfall', 'Wind_Speed', 'Soil_Moisture',
       'Soil_pH', 'Organic_Matter', 'Pest_Hotspots', 'Weed_Coverage',
       'Pest_Damage', 'Crop_Growth_Stage', 'Expected_Yield', 'Crop_Type',
       'Ground_Truth_Segmentation', 'Bounding_Boxes', 'Water_Flow',
       'Drainage_Features', 'Crop_Health_Label'],
      dtype='str')
Index(['Temperature', 'Humidity', 'Soil_Moisture', 'Crop_Health_Label'], dtype='str')


In [5]:
df.shape

(212019, 4)

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 212019 entries, 0 to 212018
Data columns (total 4 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   Temperature        212019 non-null  float64
 1   Humidity           212019 non-null  float64
 2   Soil_Moisture      212019 non-null  float64
 3   Crop_Health_Label  212019 non-null  int64  
dtypes: float64(3), int64(1)
memory usage: 6.5 MB


In [7]:
df.describe()

,Temperature,Humidity,Soil_Moisture,Crop_Health_Label
count,212019.000000,212019.000000,212019.000000,212019.000000
mean,24.988813,60.026018,20.036496,0.698985
std,4.998123,17.300364,8.657071,0.458701
min,2.683095,30.000070,5.000124,0.000000
25%,21.618165,45.077182,12.560251,0.000000
50%,24.983795,60.049335,20.062687,1.000000
75%,28.363127,74.972133,27.543781,1.000000
max,46.783110,89.999833,34.999900,1.000000


### Feature analysis

| Column            |    Mean |   Std |    Min |     Max | Interpretation                             |
| ----------------- | ------: | ----: | -----: | ------: | ------------------------------------------ |
| Temperature       | 24.99°C |  5.00 | 2.68°C | 46.78°C | Average temperature is around 25°C         |
| Humidity          |  60.03% | 17.30 | 30.00% |  90.00% | Data covers dry to highly humid conditions |
| Soil Moisture     |   20.04 |  8.66 |   5.00 |   35.00 | Soil moisture varies widely                |
| Crop Health Label |   0.699 | 0.459 |      0 |       1 | About 70% samples are healthy              |


In [8]:
df.duplicated().sum()

np.int64(0)

In [9]:
df.isnull().sum()

Temperature          0
Humidity             0
Soil_Moisture        0
Crop_Health_Label    0
dtype: int64

In [10]:
df['Crop_Health_Label'].value_counts()

Crop_Health_Label
1    148198
0     63821
Name: count, dtype: int64

In [11]:
df.to_csv(
    r"C:\IOT_Based_Crop_Disease_Detection_System_For_Smart_Agriculture\datasets\sensor_dataset\sensor_data.csv",
    index=False
)